File: flutter/lib/serverinfo.dart
import 'package:myapp/serviceinfo.dart';

class ServerInfo {
  const ServerInfo({this.name, this.services});

  final String name;

  final List<ServiceInfo> services;

  factory ServerInfo.fromJson(Map<String, dynamic> json) {
    return ServerInfo(
        name: json['name'],
        services: (json['services'] as List)
            .map((p) => ServiceInfo.fromJson(p))
            .toList());
  }
  bool running() {
    bool ok = services.isNotEmpty;
    for (ServiceInfo service in services) {
      if (!service.running) {
        ok = false;
      }
    }

    return ok;
  }

  bool notAvailable() {
    return services.isEmpty;
  }
}

File: springboot/src/main/java/com/serverchecker/dto/ServerInfo.java

package com.serverchecker.dto;

import java.util.List;

public class ServerInfo {
    
    public final String name;

    public final List<ServiceInfo> services;

    public ServerInfo(String name, List<ServiceInfo> services) {
        this.name = name;
        this.services = services;
    }
    
    
    
    
    
}
